SDKBuildScripts/Update-PlayFabSdk.ps1

```diff
@@ -0,0 +1,262 @@
+    [string]$Cloud,
+    [Parameter(ParameterSetName="ApiSpecGitUrl")]
+    [switch]$UseApiSpecFromGit,
+    [string]$OutputPath = "..\..\sdks",
+    [string]$TargetSource,
+    [switch]$KeepSource,
+    [switch]$NonNullable,
+    [switch]$Beta
+)
+
+begin
+{
+    if(!(Get-Command node))
+    {
+        throw "You must have Node.js installed to generate the SDK"
+    }
+
+    $sdkTargetSrcMap = @{
+        "ActionScriptSDK" = "actionscript";
+        "Cocos2d-xSDK" = "cpp-cocos2dx";
+        "CSharpSDK" = "csharp";
+        "JavaSDK" = "java";
+        "JavaScriptSDK" = "javascript";
+        "LuaSDK" = "LuaSdk";
+        "NodeSDK" = "js-node";
+        "Objective_C_SDK" = "objc";
+        "PhpSDK" = "PhpSdk";
+        "PostmanCollection" = "postman";
+        "PythonSDK" = "PythonSdk";
+        "SdkTestingCloudScript" = "SdkTestingCloudScript";
+        "UnrealMarketplacePlugin" = "UnrealMarketplacePlugin";
+        "UnitySDK" = "unity-v2";
+        "WindowsSDK" = "windowssdk";
+        "XPlatCppSDK" = "xplatcppsdk";
+    }
+
+    $sdksPath = $OutputPath
+    if(![System.IO.Path]::IsPathRooted($sdksPath))
+    {
+        $sdksPath = Resolve-Path (Join-Path $PSScriptRoot $OutputPath)
+    }
+
+    if(!(Test-Path $sdksPath))
+    {
+        mkdir $sdksPath | Out-Null
+    }
+
+    if($Vertical -or $Cloud)
+    {
+        # If either cloud or vertical is not provided, we want to remove the leading or trailing dot.
+        $cloudVertical = "$Vertical.$Cloud".TrimEnd(".")
+        $ApiSpecUrl = "https://$($cloudVertical).playfabapi.com/apispec"
+    }
+
+    if($ApiSpecPath)
+    {
+        if(!(Test-Path $ApiSpecPath))
+        {
+            throw "Unable to find ApiSpecPath '$ApiSpecPath'."
+        }
+
+        $apiSpecSource = "-apiSpecPath $ApiSpecPath"
+    }
+    elseif($ApiSpecUrl)
+    {
+        if(![Uri]::IsWellFormedUriString($ApiSpecUrl, "Absolute"))
+        {
+            throw "You must specify a valid URL for ApiSpecUrl.  Unable to parse '$ApiSpecUrl' as a URL."
+        }
+
+        $apiSpecSource = "-apiSpecPfUrl $ApiSpecUrl"
+    }
+    else
+    {
+        $apiSpecSource = "-apiSpecGitUrl"
+    }
+
+    Write-Verbose "API Spec Source: $apiSpecSource"
+}
+
+process
+{
+    foreach($sdkName in $SdkNames)
+    {
+        if(!$TargetSource)
+        {
+            $TargetSource = $sdkTargetSrcMap[$sdkName]
+        }
+        $destPath = Join-Path $sdksPath $sdkName
+
+        if(!(Test-Path $destPath))
+        {
+            $repoPath = "https://github.com/PlayFab/$sdkName"
+
+            if($PSCmdlet.ShouldProcess(
+                "Cloning SDK Repository for $sdkName into '$destPath'.",
+                "Would you like to clone the $sdkName repository from $($repoPath) into '$destPath'?",
+                "Unable to find $sdkName repository")
+              )
+            {
+                git clone $repoPath $destPath
+            }
+        }
+
+        # Add any overrides for a specific SDK
+        if($sdkName -eq "UE4")
```

Comment:
I think the comparison is "UnrealMarketplacePlugin", not "UE4"

Reply:
And also, you don't need an input for NonNullable at the top.
UE4 is explicitly the only non-nullable SDK right now.